java/vector/src/main/java/org/apache/arrow/vector/types/Types.java

```diff
@@ -495,7 +496,11 @@ public FieldVector getNewVector(
           Field field,
           BufferAllocator allocator,
           CallBack schemaChangeCallback) {
-        return new FixedSizeBinaryVector(field, allocator);
+        if (BinaryPointerVector.isBinaryPointerType(field.getFieldType())) {
```

Comment:
same comment, we should figure out how to avoid special casing here.  I forget what being in types gives you here.  Wouldn't you always have special case code in your own application to initialize BinaryPointerVector if needed?